allennlp_models/common/model_card.py

```diff
@@ -0,0 +1,267 @@
+import os
+import logging
+from dataclasses import dataclass
+from typing import Optional, Union, Mapping, Dict
+
+from allennlp.models import Model
+from allennlp.common.checks import ConfigurationError
+
+PRETRAINED_MODELS: Dict = {}
+STORAGE_LOCATION = "https://storage.googleapis.com/allennlp-public-models/"
+
+logger = logging.getLogger(__name__)
+
+
+def get_pretrained_models():
+    pretrained_models = {}
+    for key, val in PRETRAINED_MODELS.items():
+        pretrained_models[key] = val.to_dict()
+    return pretrained_models
+
+
+def get_description(model_class):
+    """
+    Returns the model's description from the docstring.
+    """
+    return model_class.__doc__.split("# Parameters")[0].strip()
+
+
+@dataclass
+class ModelCardInfo(object):
+    """
+    Base class for different recommended attributes included
+    in a model card.
+    """
+
+    @classmethod
+    def from_object(cls, obj: Union[str, "ModelCardInfo"]):
+        """
+        Creates the relevant `ModelCardInfo` obj. If the input is str,
+        it is initialized as the first attribute of the class.
+        """
+        if isinstance(obj, str):
+            return cls(obj)  # type: ignore
```

Comment:
Woha. So now that you have a base class that does this, you could add a `__str__` method that does the opposite, automatically. Something like
```Python
def __str__(self):
    if self.all_fields_are_str_or_none():
        return "\n\n".join(v for v in self.__dict__.values() if v is not None)
    else:
        return super(self).__str__()
```